src/util/Injected.js

```diff
@@ -403,23 +413,45 @@ exports.LoadUtils = () => {
             mediaData.type = 'document';
         }
 
+        if(mediaInfo.filesize && maxFileSize > mediaInfo.file){
+            throw new Error(`Media size ({$mediaInfo.filesize}) exceeds current upload limit ({$maxFileSize})`);
+        }
+
         if (!(mediaData.mediaBlob instanceof window.Store.OpaqueData)) {
             mediaData.mediaBlob = await window.Store.OpaqueData.createFromData(mediaData.mediaBlob, mediaData.mediaBlob.type);
         }
 
         mediaData.renderableUrl = mediaData.mediaBlob.url();
         mediaObject.consolidate(mediaData.toJSON());
         mediaData.mediaBlob.autorelease();
-
-        const uploadedMedia = await window.Store.MediaUpload.uploadMedia({
+        let mediaKeyInfoKey = await window.WWebJS.generateHash(32);
+        const mediaKeyInfoTimestamp = Date.now();
+        
+        let uploadedMedia = await window.Store.MediaUpload.uploadMedia({
             mimetype: mediaData.mimetype,
             mediaObject,
-            mediaType
+            mediaType,
+            isViewOnce: isViewOnce,
```

Comment:
You shouldn't use 
```js
isViewOnce: isViewOnce
```
You can use
```js
isViewOnce
```
Same for `uploadOrigin` and `forwardedFromWeb`